### PR TITLE
679 - Request Specific Amount screen doesn't check for the amount

### PR DESCRIPTION
**Pull request by brendoncoin**

### CODE REVIEW

File: src/js/services/wallet-address-listener.service.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+          return;
+        }
+
+        var statusStatus = 'missing';
+        if (status) {
+          statusStatus = status.isValid ? 'valid' : 'invalid';
+        }
+        console.log('walletAddressServiceListener Received status: ' + statusStatus);
+
+        var walletStatusStatus = 'missing';
```
Comment (holemcross) [inline src/js/services/wallet-address-listener.service.js]: Nothing wrong here, just funny var name.

File: src/js/services/wallet-address-listener.service.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+    function _connectSocket() {
+
+      if (currentAddressSocket) {
+        _closeSocketNormally();
+      }
+  
+      if (wallet.coin === 'bch') {
+        // listen to bch address
+        currentAddressSocket = new WebSocket('ws://47.254.143.172:80/v1/address');
```
[the remark this answers is not in the document]
Comment (brendoncoin) [reply src/js/services/wallet-address-listener.service.js]: They are only used in one place, so I don't think that adding a layer of indirection by declaring them elsewhere is worth the cost.

File: src/js/controllers/wallet-details.controller.js [hunk 3]
```diff
@@ -383,7 +383,10 @@ angular.module('copayApp.controllers').controller('walletDetailsController', fun
 
     $scope.walletId = data.stateParams.walletId;
     $scope.wallet = profileService.getWallet($scope.walletId);
+    //console.log('walletDetailsController beforeEnter wallet not null: ' + $scope.wallet !== null);
```
[the remark this answers is not in the document]
Comment (brendoncoin) [reply src/js/controllers/wallet-details.controller.js]: Removed.

File: src/js/services/wallet-address-listener.service.js (hunk 1 above)
Comment (brendoncoin) [reply src/js/services/wallet-address-listener.service.js]: Removed debug code.

File: src/js/controllers/custom-amount.controller.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+        var satoshisReceived = 0;
+        data.outputs.forEach(function onOutput(output) {
+          if (output.address === listeningAddressLegacy) {
+            satoshisReceived += output.value;
+          }
+        });
+
+        vm.showingPaymentReceived = satoshisReceived === satoshisRequested;
+        console.log('satoshisReceived: ', satoshisReceived);
```
[the remark this answers is not in the document]
Comment (brendoncoin) [reply src/js/controllers/custom-amount.controller.js]: Removed.

**Pull request merged**